src/IAllocatorConduit.sol

```diff
@@ -0,0 +1,190 @@
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.13;
+
+/**
+ *  @title IAllocatorConduit
+ *  @dev   Conduits are to be used to manage positions for multiple Allocators.
+ *         After funds are deposited into a Conduit, they can be deployed by Fund Managers to earn yield.
+ *         When Allocators want funds back, they can request funds from the Fund Managers and then withdraw once liquidity is available.
+ */
+interface IAllocatorConduit {
+
+    /**
+     *  @dev   Event emitted when a deposit is made to the Conduit.
+     *  @param domain The unique identifier of the domain.
+     *  @param asset     The address of the asset deposited.
+     *  @param amount    The amount of asset deposited.
+     */
+    event Deposit(bytes32 indexed domain, address indexed asset, uint256 amount);
+
+    /**
+     *  @dev   Event emitted when a withdrawal is made from the Conduit.
+     *  @param domain   The unique identifier of the domain.
+     *  @param asset       The address of the asset withdrawn.
+     *  @param destination The address where the asset is sent.
+     *  @param amount      The amount of asset withdrawn.
+     */
+    event Withdraw(bytes32 indexed domain, address indexed asset, address destination, uint256 amount);
+
+    /**
+     *  @dev   Event emitted when a Conduit request is made.
+     *  @param domain     The unique identifier of the domain.
+     *  @param asset         The address of the asset to be withdrawn.
+     *  @param amount        The amount of asset to be withdrawn.
+     *  @param data          Arbitrary encoded data to provide additional info to the Fund Manager.
+     *  @param fundRequestId The ID of the fund request.
+     */
+    event RequestFunds(bytes32 indexed domain, address indexed asset, uint256 amount, bytes data, uint256 fundRequestId);
+
+    /**
+     *  @dev   Event emitted when a fund request is cancelled.
+     *  @param domain     The unique identifier of the domain.
+     *  @param asset         The address of the asset for the cancelled request.
+     *  @param amount        The amount of asset for the cancelled request.
+     *  @param data          Arbitrary encoded data to provide additional info to the Fund Manager.
+     *  @param fundRequestId The ID of the cancelled fund request.
+     */
+    event CancelRequest(bytes32 indexed domain, address indexed asset, uint256 amount, bytes data, uint256 fundRequestId);
+
+    /**
+     *  @dev   Event emitted when a fund request is filled.
+     *  @param domain The unique identifier of the domain.
+     *  @param asset     The address of the asset for the filled request.
+     *  @param amount    The amount of asset for the filled request.
+     *  @param data      Arbitrary encoded data to provide additional info to the Conduit.
+     */
+    event FillRequest(bytes32 indexed domain, address indexed asset, uint256 amount, bytes data);
```

Comment:
In which function and exactly how this event is being emitted?

Reply:
@lucas-manuel ^

Reply:
The idea is that this is fired when fund requests are met. I think we may need to take another iteration on this whole process now that we have 3 implementations taking form. The Spark implementation for example does not support requests and instead the fund requests are pooled. For the Spark case we probably don't want to fire this event.